src/Keyable.php

```diff
@@ -11,8 +12,17 @@ public function apiKeys()
         return $this->morphMany(ApiKey::class, 'keyable');
     }
 
-    public function createApiKey()
+    public function createApiKey(array $attributes = []): NewApiKey
     {
-        return $this->apiKeys()->create([]);
+        $planTextApiKey = ApiKey::generate();
+
+        $apiKey = Model::withoutEvents(function () use ($planTextApiKey, $attributes) {
```

Comment:
Curious about the reason for suppressing model events 🤔

Reply:
Because it would get hashed again in `static::creating` inside `ApiKey` model.